src/as_execution/abi.rs

```diff
@@ -864,16 +1199,36 @@ pub(crate) fn assembly_script_send_message(
     };
 
     env.get_interface().send_message(
-        target_address,
-        target_handler,
+        &target_address,
+        &target_handler,
         validity_start,
         validity_end,
         max_gas as u64,
         raw_fee as u64,
         raw_coins as u64,
-        data,
+        &data,
         filter,
     )?;
+
+    #[cfg(feature = "execution-trace")]
+    ctx.data_mut().trace.push(AbiTrace {
+        name: function_name!().to_string(),
+        params: vec![
+            into_trace_value!(target_address),
+            into_trace_value!(target_handler),
+            into_trace_value!(validity_start),
```

Comment:
Maybe better to match the parameters of the abi exactly and so split the slot in 2 differents params

Reply:
Done